### PR TITLE
add slack alert implementation

### PR DESCRIPTION
**Pull request by robpickerill**
This PR adds slack as a notifier and alerting mechanisms for probes.

### CODE REVIEW

File: probes/alerting/notifier/slack/slack.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+// New creates a new Slack client.
+func New(slackcfg *configpb.Slack, l *logger.Logger) (*Client, error) {
+	webhookURL, err := lookupWebhookUrl(slackcfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		httpClient: &http.Client{},
```
Comment (robpickerill) [inline probes/alerting/notifier/slack/slack.go:37]: I wasn't sure if creating http clients for each notifier for pd and slack is scalable, I can look at threading a http client instead if not

File: probes/alerting/notifier/slack/slack.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		return fmt.Errorf("Slack webhook returned status code %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+// createMessage creates a new Slack webhook message, from the alertFields
+func createMessage(alertFields map[string]string) webhookMessage {
+	return webhookMessage{
+		Text: alertFields["details"],
```
Comment (robpickerill) [inline probes/alerting/notifier/slack/slack.go:108]: I was unsure how to format the message, I'd thought about defining a slack template and passing it through `strtemplate`, but decided to take the details template for now and rework later

File: probes/alerting/notifier/slack/slack.go (hunk 1 above)
Comment (manugarg) [reply probes/alerting/notifier/slack/slack.go:37]: I think it's okay.

File: probes/alerting/notifier/slack/slack.go (hunk 2 above)
Comment (manugarg) [reply probes/alerting/notifier/slack/slack.go:108]: Sounds good as details is already parsed through strtemplate.

**Review by manugarg — APPROVED** [review]
Looks good to me.

**Comment by manugarg** [comment]
@robpickerill, looks like my changes caused some conflict, I am not able to merge. Can you please resolve the conflicts and push again?

**Comment by robpickerill** [comment]
> please resolve the conflicts

no problems, should be good to merge now, thanks

**Pull request merged**